opengrok-indexer/src/main/java/org/opengrok/indexer/util/FileExtraZipper.java

```diff
@@ -75,9 +74,11 @@ private FileExtra findExtra(Map<String, FileExtra> byName, File fileobj) {
     private Map<String, FileExtra> indexExtraByName(List<FileExtra> extras) {
         Map<String, FileExtra> byPath = new HashMap<>();
         for (FileExtra extra : extras) {
-            File f = new File(extra.getFilepath());
-            String filename = f.getName();
-            byPath.put(filename, extra);
+            if (extra.getFilepath() != null) {
```

Comment:
this I added because of interaction with index created with the #3138 changes - was getting NPEs due to `filepath = null`.